Tests/TutorLizard.BusinessLogic.Tests/Services/Student/StudentServiceScheduleItemRequestTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Moq;
+using TutorLizard.BusinessLogic.Interfaces.Data.Repositories;
+using TutorLizard.BusinessLogic.Models;
+using TutorLizard.BusinessLogic.Models.DTOs.Requests;
+using TutorLizard.BusinessLogic.Services;
+
+namespace TutorLizard.BusinessLogic.Tests.Services.Student
+{   
+    public class StudentServiceScheduleItemRequestTests : StudentServiceTestBase
+    {
+        [Fact]
+        public async Task CreateScheduleItemRequest_WhenIsRemoteIsTrue_ShouldSetIsRemoteToTrue()
+        {
+            // Arrange
+            var scheduleItem = new ScheduleItem { Id = 1, Ad = new Ad { TutorId = 2 } };
+            SetupMockGetScheduleItemById(scheduleItem);
+
+            var request = new CreateScheduleItemRequestRequest
+            {
+                StudentId = 2,
```

Comment:
You set  `StudentId` the same as `TutorId`. This test should not pass, because the service would return failed response (the `isOwner` would be true in `StudentService` line 38) before it would get to creating the request.

The reason this test passes, is that you set up only the `GetById` (which isn't used in the tested method), but not `GetAll`. I'm actually surprised that when the service calls `GetAll` to check if student is the owner it gets false instead of an exception.
Whether the method calls `GetById` or `GetAll` is an implementation detail and when testing, you should be prepared for both these cases. You somehow arrived at a positive outcome, but in our actual app, you would get a different execution path.

Setting up `GetAll` requires you to add the `Ad` attached to the `ScheduleItem` to be added to the fake DB, which doesn't work until you provide all required fields. I'd recommend adding a protected method to `StudentServiceTestBase` that creates a test `Ad` using `Fixture` (just remember to do `.Without()` on any navigation property).

Reply:
You set up `GetAll` in one `CreateScheduleItemRequest_WhenIsRemoteIsTrue_ShouldSetIsRemoteToTrue`, but left it as it was in `CreateScheduleItemRequest_WhenIsRemoteIsFalse_ShouldSetIsRemoteToFalse` (I still think it would be easier to maintain if it was one test with 2 cases) and in `CreateScheduleItemRequest_WhenRequestSent_ShouldReturnSuccess` as well. 
What I'm trying to say, is you need to do this setup for any test of this method. Maybe create another private method that runs both `GetAll` and `GetById` setups?